pkg/plugins/manager/client/client.go

```diff
@@ -204,3 +218,73 @@ func (s *Service) plugin(ctx context.Context, pluginID string) (*plugins.Plugin,
 
 	return p, true
 }
+
+// removeConnectionHeaders removes hop-by-hop headers listed in the "Connection" header of h.
+// See RFC 7230, section 6.1
+//
+// Based on https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/net/http/httputil/reverseproxy.go;l=411-421
+func removeConnectionHeaders(h map[string][]string) {
+	for _, f := range h["Connection"] {
+		for _, sf := range strings.Split(f, ",") {
+			if sf = textproto.TrimString(sf); sf != "" {
+				delHeader := ""
+				for k := range h {
+					if textproto.CanonicalMIMEHeaderKey(sf) == textproto.CanonicalMIMEHeaderKey(k) {
+						delHeader = k
+						break
+					}
+				}
+
+				if delHeader != "" {
+					delete(h, delHeader)
+				}
+			}
+		}
+	}
+}
+
+// Hop-by-hop headers. These are removed when sent to the backend.
+// As of RFC 7230, hop-by-hop headers are required to appear in the
+// Connection header field. These are the headers defined by the
+// obsoleted RFC 2616 (section 13.5.1) and are used for backward
+// compatibility.
+//
+// Copied from https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/net/http/httputil/reverseproxy.go;l=171-186
+var hopHeaders = []string{
+	"Connection",
+	"Proxy-Connection", // non-standard but still sent by libcurl and rejected by e.g. google
+	"Keep-Alive",
+	"Proxy-Authenticate",
+	"Proxy-Authorization",
+	"Te",      // canonicalized version of "TE"
+	"Trailer", // not Trailers per URL above; https://www.rfc-editor.org/errata_search.php?eid=4522
+	"Transfer-Encoding",
+	"Upgrade",
+}
+
+// removeHopByHopHeaders removes hop-by-hop headers. Especially
+// important is "Connection" because we want a persistent
+// connection, regardless of what the client sent to us.
+//
+// Based on https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/net/http/httputil/reverseproxy.go;l=276-281
+func removeHopByHopHeaders(h map[string][]string) {
+	for _, hh := range hopHeaders {
+		delHeader := ""
+		for k := range h {
+			if hh == textproto.CanonicalMIMEHeaderKey(k) {
+				delHeader = k
+				break
+			}
+		}
+
+		if delHeader != "" {
+			delete(h, delHeader)
+		}
+	}
+}
```

Comment:
I think it'd be simpler to do: 

```suggestion
func removeHopByHopHeaders(h map[string][]string) {
	for _, hh := range hopHeaders {
		for k := range h {
			if hh == textproto.CanonicalMIMEHeaderKey(k) {
				delete(h, k)
				break
			}
		}
	}
}
```

Same situation for the connection headers

Reply:
Hmm is it really a good advice to delete a key in a map while you're iterating it?

Reply:
Oh seems like it is. I'll change

Reply:
Yeah good old Golang!